lib/promoted/ruby/client/faraday_http_client.rb

```diff
@@ -10,8 +10,8 @@ def initialize
                 @conn = Faraday.new do |f|
                     f.request :json
                     f.request :retry, max: 3
-                    f.adapter :net_http
                     f.use Faraday::Response::RaiseError # raises on 4xx and 5xx responses
+                    f.adapter :net_http
```

Comment:
I'm okay with this.  I was thinking that, eventually, it might be good to not initialize the Faraday client unless one of the server endpoints were specified.  However, we almost always want clients to set the endpoints so my thoughts might not be useful on it.

Reply:
Yes, the next thing I would do, in fact, is make the client pluggable.